rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/uri/FallbackResolver.java

```diff
@@ -250,5 +272,45 @@ public void unwatch(ISourceLocation root, Consumer<ISourceLocationChanged> watch
         getClient().removeWatcher(root, watcher, getServer());
 
     }
+    
+    public boolean isFileManaged(ISourceLocation file) {
+        for (final var service : textDocumentServices) {
+            if (service.isManagingFile(file)) {
+                return true;
+            }
+        }
+        return false;
+    }
 
+    @Override
+    public ISourceLocation resolve(ISourceLocation input) throws IOException {
+        if (isFileManaged(input)) {
+            try {
+                return URIUtil.changeScheme(input, "lsp+" + input.getScheme()).top();
+            } catch (URISyntaxException e) {
+                // fall through
+            }
+        }
+        return input;
+    }
+
+    @Override
+    public String authority() {
+        throw new UnsupportedOperationException("'authority' not supported by fallback resolver");
+    }
+
+    private Set<IBaseTextDocumentService> textDocumentServices = new HashSet<>();
+
+    public void registerTextDocumentService(IBaseTextDocumentService service) {
+        textDocumentServices.add(service);
+    }
+
+    public TextDocumentState getDocumentState(ISourceLocation file) {
+        for (final var service : textDocumentServices) {
+            if (service.isManagingFile(file)) {
+                return service.getDocumentState(file);
+            }
+        }
+        return null;
```

Comment:
is there an alternative to `null`? should we throw an exception at this point? since `isFileManaged` is true?

what happens to a uri that was generated while the editor was open, but refenced after it is closed? do we unresolve it to the original URI?